fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/UploadTaskRequest.kt

```diff
@@ -14,8 +14,8 @@ class UploadTaskRequest(request: Request) : TaskRequest(request) {
 
     val BUFFER_SIZE = 1024
 
-    val CRLF = "\\r\\n"
-    val boundary = System.currentTimeMillis().toHexString()
+    val CRLF = "\r\n"
+    val boundary = request.httpHeaders["Content-Type"]!!.split("=", limit=2)[1]
```

Comment:
The `!!` here is intended to mean that "Content-Type" must be here or something terrible has happed.

I guess a more verbose error is prefered, but I do not know kotlin very well and am not sure how to do that.